### PR TITLE
Add LockBranch and AllowForkSyncing to repos.go

### PR DESCRIPTION
**Pull request by XaurDesu**
Fixes https://github.com/google/go-github/issues/2574 , and it is mostly a direct, if barebones, implementation of the plan ideated there. Adding references to `LockBranch` and `AllowForkSyncing`. References to `RequireLastPushApproval` were added in cb24cabfacd4bff06727d5b3acea48ba56e5f690 . There might be some changes left to do.

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -894,11 +896,21 @@ type AdminEnforcedChanges struct {
 	From *bool `json:"from,omitempty"`
 }
 
+// LockBranch represents if the branch is marked as read-only. If this is true, users will not be able to push to the branch.
+type LockBranch struct {
+	From *bool `json:"lock_branch"`
+}
+
```
Comment (gmlewis) [inline github/repos.go:903]: Please delete these lines.

File: github/repos.go [hunk 2]
```diff
@@ -840,6 +840,8 @@ type Protection struct {
 	AllowForcePushes               *AllowForcePushes               `json:"allow_force_pushes"`
 	AllowDeletions                 *AllowDeletions                 `json:"allow_deletions"`
 	RequiredConversationResolution *RequiredConversationResolution `json:"required_conversation_resolution"`
+	LockBranch                     *LockBranch                     `json:"lock_branch"`
+	AllowForkSyncing               *AllowForkSyncing               `json:"allow_fork_syncing"`
```
Comment (gmlewis) [inline github/repos.go:844]:
```suggestion
	// LockBranch represents if the branch is marked as read-only. If this is true, users will not be able to push to the branch.
	LockBranch                     *bool                     `json:"lock_branch,omitempty"`
	// AllowForkSyncing represents whether users can pull changes from upstream when the branch is locked.
	AllowForkSyncing               *bool               `json:"allow_fork_syncing,omitempty"`
```

File: github/repos.go [hunk 3]
```diff
@@ -894,11 +896,21 @@ type AdminEnforcedChanges struct {
...
 // AllowDeletionsEnforcementLevelChanges represents the changes made to the AllowDeletionsEnforcementLevel policy.
 type AllowDeletionsEnforcementLevelChanges struct {
 	From *string `json:"from,omitempty"`
 }
 
+// AllowForkSyncing represents whether users can pull changes from upstream when the branch is locked.
+type AllowForkSyncing struct {
+	From *bool `json:"allow_fork_syncing"`
+}
+
```
Comment (gmlewis) [inline github/repos.go:913]: Please delete these lines.

**Review by gmlewis — CHANGES_REQUESTED** [review]
After making the requested changes below, please run `gofmt` on `repos.go` and then also run `go generate ./...` at the top of the repo and push (not force-push) any changes that were made, as described in the file CONTRIBUTING.md.  
Thanks.

File: github/repos.go (hunk 1 above)
Comment (XaurDesu) [reply github/repos.go]: Just commited a deletion for both struct

**Comment by XaurDesu** [comment]
I can't seem to replicate the failing test in my local environment, already ran `gofmt` and `go generate`, but it seems to have no effect on tests, i think i might have a configuration problem on my end but i can't seem to figure it out.

**Comment by dtdao** [comment]
From checking the workflow file. The tests didnt run.  Its failing at the
https://github.com/google/go-github/blob/621c4ba7155bee18761f308a86fa71fddf3476b4/.github/workflows/tests.yml#L64

specifically at `git diff --exit-code; ` portion of it.

**Comment by gmlewis** [comment]
Yes, you are correct, @dtdao - thank you for the detailed explanation. 

This information should also be covered in our CONTRIBUTING.md guide at the top of this repo... if you feel that it can be made clearer, we welcome PRs for it as well.

**Comment by XaurDesu** [comment]
I agree that `CONTRIBUTING.md` could (and probably should) be clearer about some things, but as it should be evident by now, I'm pretty new to this kind of workflow, and i would rather solve a few more issues to get accustomed to the codebase before messing with any documentation we have, especially kind that regards new contributors. Maybe @dtdao could check that out.

File: github/repos.go (hunk 3 above)
Comment (XaurDesu) [reply github/repos.go]: done.

**Review by gmlewis — APPROVED** [review]
Thank you, @XaurDesu !
LGTM.
Merging.

**Pull request merged**